File: src/main/java/com/mpe85/grampa/input/impl/StringSegmentInputBuffer.java
package com.mpe85.grampa.input.impl;

import com.ibm.icu.impl.StringSegment;
import com.mpe85.grampa.input.InputBuffer;

/**
 * An {@link InputBuffer} implementation using a {@link StringSegment}.
 * 
 * @author mpe85
 *
 */
public class StringSegmentInputBuffer extends CharSequenceInputBuffer {
	
	private final StringSegment stringSegment;
	
	public StringSegmentInputBuffer(final StringSegment stringSegment) {
		super(stringSegment);
		this.stringSegment = stringSegment;
	}
	
	@Override
	public int getCodePoint(final int index) {
		return stringSegment.codePointAt(index);
	}
	
}

File: src/main/kotlin/com/mpe85/grampa/input/impl/StringSegmentInputBuffer.kt
package com.mpe85.grampa.input.impl

import com.ibm.icu.impl.StringSegment
import com.mpe85.grampa.input.InputBuffer

/**
 * An [InputBuffer] implementation using a [StringSegment].
 *
 * @author mpe85
 * @property[stringSegment] A string segment backing the input buffer
 */
public class StringSegmentInputBuffer(private val stringSegment: StringSegment) :
    CharSequenceInputBuffer(stringSegment) {

    override fun getCodePoint(index: Int): Int = stringSegment.codePointAt(index)
}
